src/Compilers/Core/CodeAnalysisTest/Collections/List/IEnumerable.Generic.Tests.cs

```diff
@@ -584,9 +584,9 @@ public void IEnumerable_Generic_Enumerator_Current_ReturnsSameObjectsOnDifferent
             Dictionary<T, int> firstValues = new Dictionary<T, int>(count);
             Dictionary<T, int> secondValues = new Dictionary<T, int>(count);
             foreach (T item in enumerable)
-                firstValues[item] = firstValues.ContainsKey(item) ? firstValues[item]++ : 1;
+                firstValues[item] = firstValues.TryGetValue(item, out var counter) ? ++counter : 1;
```

Comment:
fixed to count correctly

Reply:
Behavior changes need to be accompanied by a test.

Reply:
So i should write a test for a test?

Reply:
Ah, I didn't realize it was a test. But if this behavior change didn't cause the test to fail, then something else is wrong

Reply:
the enumerable only contains unique items, so it never hits that code path
https://github.com/dotnet/roslyn/blob/e1ee2f544a7a4f8d536278bed4e180c54919276e/src/Compilers/Core/CodeAnalysisTest/Collections/List/IEnumerable.Generic.Tests.cs#L33-L35

Reply:
Is this true for every one of these refactorings (in this file and below)? If so, I think we should just remove the dead code.

Reply:
The non generic should be the same.
But its only dead code when the implementation of enumerable behaves correctly, if a second call would repeat the first element twice than it would now correctly fail the test.

Reply:
Sure, but that just means we can fail the test earlier if there's a repeat.